### DISCUSSION TITLE
Conflicting color usage for issue status indicator

### DISCUSSION THREAD

**Discussion by AlexV525** [community/community#6795] [Projects and Issues]
This is a feedback targeting the roadmap issue: https://github.com/github/roadmap/issues/289#issue-1036472304 .

(I've also filed a ticket for this.)

### Summary

Purple should only use when the issue is explicitly marked as *Resolved*. A closed issue could be *accidently opened*, *spam*, *out of topic*, or something else rather than *won't fix* and *duplicate*.

*A good upgrade experiences should be opt-in, not opt-out*, which means options can be added to the "Close issue" button like "Close as resolved", to explicitly marked as merged if it's not closed by a merged PR.

Examples:
- https://github.com/dart-lang/pub-dev/issues/5078 (wrong repo)
- https://github.com/flutter/website/issues/6166 (discontinued)
- https://github.com/flutter/flutter-intellij/issues/5469 (moved to another repo)

(As I writing this, I can still see the red indicator for closed issues in this discussion's preview.)

Issues that not closed by PR has an misleading CSS style (`State--merged`) too,
![image](https://user-images.githubusercontent.com/15884415/139004906-153523d5-d047-4468-92b3-4c17be65cca4.png)

### Proposal/Solutions

- Revert the change (which seems impossible to be done).
- Split the closing status more precisely, with closed and resolved.
- Use more colors to indicate.

To be honest, how can you tell these issues are closed but not resolved if there are no such words saying "closed"?
![image](https://user-images.githubusercontent.com/15884415/139005348-cbc7d99d-95ad-44c7-a747-695f4a130a9a.png)

**Comment by timoyates** [comment]
Hi @AlexV525, thanks for the feedback! It was very insightful and has been shared with the product team. Cheers!

**Reply by AlexV525** [reply]
> Hi @AlexV525, thanks for the feedback! It was very insightful and has been shared with the product team. Cheers!

@timoyates Thanks for noticing this. State misleading should not happen in such great projects, at least it should be opt-in.

**Reply by davendu** [reply]
Just **change the color back**.

In the roadmap you guys said "We will provide the capability to ... differentiate between closing because it is not going to be worked on or closing because it is done" then change the color back until you have this capability. You can even see from the words that, it require more letters to describe the difference between a "done" issue, and a "closed" issue.

Why not add a new field to indicate this difference? Why punishing our mental health?

**Comment by Katsute** [comment]
This sudden change from red to purple should've been in the **feature preview** before it was pushed to the public, personally I am not a big fan of this change.

I've created a quick chrome extension if anyone still prefers the old closed-red color: <https://github.com/Katsute/GitHub-Red-Closed-Issues-Extension>

**Reply by timoyates** [reply]
Thanks for the feedback, @Katsute. We'll be continuing to iterate on the color and states over time and will keep it in mind as we do so. Cheers!

**Comment by TheBrokenRail** [comment]
*Copying from #6801*

I agree that the new closed issue color needs to be reverted back to red (or add an equivalent of the merged state for issues).

Pull requests have 3 main states, represented by 3 main colors: open is green, closed is red, and merged is purple. And as such, it is quite easy to start associating these colors with their states, it makes it easy to tell at a glance what state something is in.

And this carries over to issues, except the merge state is dropped. Or at least it did carry over to issues. Now that purple represents both merged and closed depending on the situation, that mental heuristic is worse than useless.

And that is why I believe the change should be reverted (or implemented in a different way).

**Comment by peppy** [comment]
Please consider rolling this back - at least temporarily - pending further consideration. The confusion it is causing for myself and my team is immense. Changing colours that were clearly indicating a status to no longer differentiate slows down issue/PR workflow and pains the brain.

**Reply by peppy** [reply]
Checking back in a week later to confirm this isn't just bias against not accommodating to change (a human trait). It's still confusing and frustrating me on a daily basis. I almost get a sense of hope when I find a usage of red that wasn't updated, but turns out it's just inconsistencies with how seemingly haphazardly this was implemented:

![20211105 124225 (Safari)](https://user-images.githubusercontent.com/191335/140454760-969e2909-5e48-47f2-afa6-ebaf68555032.png)

**Comment by atom0s** [comment]
This just feels like another rushed design choice with no actual purpose. Change for the sake of change is not good. Please stop doing it.

As others have stated, this should have been a beta'd feature with your preview system. Because this is something that would have been immediately turned off/unwanted by everyone who tested it.

Also, if you open your statements with things like:

> Our current issue icon colors are a source of constant user feedback, citing confusion with errors, confusion between why Open is green and Closed is red, accessibility concerns and the general scariness of seeing red across the issues index page when a bunch of closed issues is usually a good thing.

Please show us the proof and numbers to back this up. I have never once seen someone complain about the colors of these icons/buttons; ever. The only time I've ever seen anyone mention color changes on GitHub were for suggestions to those who have color blindness. _(Which is a totally understandable suggestion for change, but as something to opt-into if you need it.)_

These new colors just make it seem like the entire page is accepted and merged pull requests. 
This is not useful or better at all.


And to further prove the point; this is exactly how it was changed/added too... why does this not use separated selectors/classes? 

A closed issue is now marked with:

```html
<span title="Status: Closed" data-view-component="true" class="State State--merged">
```

Even though it's just closed, with no related pull or anything.

**Comment by bdach** [comment]
Adding to the above feedback, in some cases the new purple usage is actively misleading. See the following pull request thread: https://github.com/ppy/osu/pull/15378

I closed the pull due to it being obviously bogus, but at the end of the thread, the action of me closing the pull *without merging* is shown with a *purple* icon. I had to have a double take to make sure I didn't *accidentally merge* this change, because merging it would result in the same colour of icon.

**Reply by timoyates** [reply]
This is excellent feedback and not the intended behavior - we will look into what's causing the "issue closed" icon displaying for closed pull requests and address it. Thanks a lot for the report and cheers!

**Comment by AlexV525** [comment]
And 1 month after this release, no further actions have been taken from the team. Are you trying to pretend that nothing happened?

**Comment by Katsute** [comment]
A Chrome/Firefox extension is available for anyone who wants to revert this change https://github.com/Katsute/GitHub-Red-Issues#readme

**Comment by j0siepy** [comment]
A folks! A fix has gone out for this! See [this discussion](https://github.com/orgs/community/discussions/6767) 💟

**Reply by AlexV525** [reply]
Sorry but the fix is only for PRs, not including issues.

**Comment by MakePixelsWork** [comment]
As of [October 26, 2021](https://github.blog/changelog/2021-10-26-updates-to-our-issue-status-icons-and-colors/) the closed issues became purple in a lot of GitHub products. But there is still some red to be found. 

![image](https://github.com/community/community/assets/56885246/a8107213-c8a7-46e5-90e5-e3da7395cf8f)

Can these closed issues be changed to purple too? Or does this conflict with the fact that 'Closed as not planned', 'Won't fix' and 'Duplicate' are shown as gray?

Its an long outstanding question among the community, but you updated them to purple almost 2.5 years ago. Shouldn't it be implemented everywhere?!

**Comment by MakePixelsWork** [comment]
Almost a year later and this is still an ongoing annoyance. Can't GitHub change this behavior?!
![image](https://github.com/user-attachments/assets/0007177e-d957-4235-a3cc-317f378e9e70)
